v2/internal/pkg/release/local_stored_collector.go

```diff
@@ -233,26 +233,22 @@ func (o LocalStorageCollector) prepareD2MCopyBatch(log clog.PluggableLoggerInter
 			o.Log.Error("%s", err.Error())
 			return nil, err
 		}
+		src = imgSpec.ReferenceWithTransport
 		if imgSpec.IsImageByDigest() {
-			src = dockerProtocol + strings.Join([]string{o.LocalStorageFQDN, imgSpec.PathComponent + "@" + imgSpec.Algorithm + ":" + imgSpec.Digest}, "/")
-			dest = strings.Join([]string{o.Opts.Destination, imgSpec.PathComponent + "@" + imgSpec.Algorithm + ":" + imgSpec.Digest}, "/")
+			tag := fmt.Sprintf("%s-%s", imgSpec.Algorithm, imgSpec.Digest)
```

Comment:
@aguidirh - simple but effective, just one question will be sure that formatted tag will always be less than the 128 character limit as indicated in the tag spec ?

Reply:
There is not guarantee, is it better to add a length validation then?

Reply:
@aguidirh - yes  I think so, just to make sure